src/sources/exec/mod.rs

```diff
@@ -379,27 +373,56 @@ async fn run_command(
         .ok_or_else(|| Error::new(ErrorKind::Other, "Unable to take stdout of spawned process"))?;
 
     // Create stdout async reader
-    let stdout = stdout.allow_read_until(shutdown.clone().map(|_| ()));
+    //let stdout = stdout.allow_read_until(shutdown.clone().map(|_| ()));
     let stdout_reader = BufReader::new(stdout);
 
     let pid = child.id();
 
     spawn_reader_thread(stdout_reader, decoder.clone(), STDOUT, sender);
 
-    while let Some(((mut events, _byte_size), stream)) = receiver.recv().await {
-        let count = events.len();
-        emit!(&ExecEventsReceived {
-            count,
-            command: config.command_line().as_str(),
-            byte_size: events.size_of(),
-        });
+    'outer: loop {
+        tokio::select! {
+            _ = &mut shutdown => {
+                dbg!("shutting down");
+                match pid.map(|pid| i32::try_from(pid)) {
+                    Some(Ok(pid)) => {
+                        // shutting down, send a SIGTERM to the child
+                        if let Err(error) = signal::kill(Pid::from_raw(pid), Signal::SIGTERM) {
+                            emit!(&ExecFailedToSignalChild{command: &command, error: ExecFailedToSignalChildError::SignalError(error)});
```

Comment:
Double check the indentation/spacing too. `rustfmt` sometimes fails inside macro blocks.